elasticsearch/src/main/java/org/apache/metamodel/elasticsearch/ElasticSearchDateConverter.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.metamodel.elasticsearch;
+
+import org.apache.metamodel.util.TimeComparator;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * Util class to convert date strings from ElasticSearch to
+ * proper java Dates.
+ */
+final class ElasticSearchDateConverter {
+
+    public static Date tryToConvert(String dateAsString) {
+        try {
+            SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSX");
+            return dateFormat.parse(dateAsString);
+        } catch (ParseException e) {
+            return TimeComparator.toDate(dateAsString);
```

Comment:
It's an improvement to add it, but I still would like to ask out (also to others?) what is best to do here ... In case we cannot parse the date format, then I tend to suggest that we simply return the dateAsString (so the method must return Object). Reason is that otherwise we might simply throw away date and that can give users the wrong impression, thinking that there is no date value in the data itself and fueling wrong application-logic then. Another approach is to throw an exception. In such a case at least the user knows that something went wrong. What I don't like is to let the user think everything is fine but that the data is null.